src/main/java/net/neoforged/neoforge/registries/datamaps/NeoForgeDataMaps.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.registries.datamaps;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.ExtraCodecs;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.level.block.ComposterBlock;
+import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.neoforge.internal.versions.neoforge.NeoForgeVersion;
+
+/**
+ * Holds all {@link DataMapType data maps} provided by NeoForge.
+ * <p>
+ * These data maps are usually replacements for vanilla in-code maps, and are optionally
+ * synced so that mods can use them on the client side.
+ */
+public class NeoForgeDataMaps {
+    /**
+     * The {@linkplain Item} data map that replaces {@link ComposterBlock#COMPOSTABLES}.
+     * <p>
+     * The location of this data map is {@code neoforge/data_maps/item/compostables.json}, and the values are objects with 1 field:
+     * <ul>
+     * <li>{@code chance}, a float between 0 and 1 (inclusive) - the chance that the item will add levels to the composter when composted</li>
+     * </ul>
+     *
+     * The use of a float as the value is also possible, though discouraged in case more options are added in the future.
+     */
+    public static final DataMapType<Item, Compostable> COMPOSTABLES = DataMapType.builder(
+            id("compostables"), Registries.ITEM, Compostable.CODEC).synced(Compostable.CHANCE_CODEC, false).build();
+
+    /**
+     * Data map value for {@linkplain #COMPOSTABLES compostables}.
+     *
+     * @param chance the chance that a compost is successful
+     */
+    public record Compostable(float chance) {
```

Comment:
I'm not a fan of making the data map value classes inner classes, I would prefer them to be at the top level in a `builtin` or `neo` subpackage

Reply:
This is the design we've accepted for other objects like built-in biome modifiers.